### PR TITLE
Log the trace id and span id in http filters

### PR DESCRIPTION
**Pull request by madhuravi**
Set trace id and span id as logger fields in the http filter chain.

### CODE REVIEW

**Comment by madhuravi** [comment]
@uber-go/service-framework

File: modules/uhttp/filters.go [hunk 1]
```diff
@@ -68,11 +69,16 @@ func tracingServerFilter(host service.Host) FilterFunc {
...
 			fxctx.Logger().Info("Malformed inbound tracing context: ", "error", err.Error())
 		}
-
 		span := opentracing.GlobalTracer().StartSpan(operationName, ext.RPCServerOption(spanCtx))
 		ext.HTTPUrl.Set(span, r.URL.String())
 		defer span.Finish()
 
+		jSpanCtx, ok := span.Context().(jaeger.SpanContext)
+		if ok {
+			fxctx.Logger().Info(
```
Comment (anuptalwalkar) [inline modules/uhttp/filters.go]: I thought this task meant to add it to the logger (using With()), and not actually output the log, so that whenever user calls info or error, Logger would populate with these preset fields.

Comment (madhuravi) [inline modules/uhttp/filters.go]: Check description. This is not directly related to the task. This is for debugging.

File: internal/fxcontext/context.go [hunk 2]
```diff
@@ -58,15 +58,28 @@ func New(ctx gcontext.Context, host service.Host) fx.Context {
 // Logger returns context based logger. If logger is absent from the context,
 // the function updates the context with a new context based logger
 func (c *Context) Logger() ulog.Log {
-	return c.getStore().log
+	store := c.getStore()
+	if store.log == nil {
```
Comment (alsamylkin) [inline internal/fxcontext/context.go]: why do you want to always check for nil? I thought store is an internal type..

File: modules/uhttp/filters.go [hunk 3]
```diff
@@ -68,13 +69,19 @@ func tracingServerFilter(host service.Host) FilterFunc {
 		if err != nil && err != opentracing.ErrSpanContextNotFound {
 			fxctx.Logger().Info("Malformed inbound tracing context: ", "error", err.Error())
 		}
-
 		span := opentracing.GlobalTracer().StartSpan(operationName, ext.RPCServerOption(spanCtx))
 		ext.HTTPUrl.Set(span, r.URL.String())
 		defer span.Finish()
 
+		jSpanCtx, ok := span.Context().(jaeger.SpanContext)
+		if ok {
```
Comment (alsamylkin) [inline modules/uhttp/filters.go]: Nit: merge with the line above?

File: modules/uhttp/filters.go [hunk 4]
```diff
@@ -68,13 +69,19 @@ func tracingServerFilter(host service.Host) FilterFunc {
...
+			traceLogger := fxctx.Logger().With(
+				"trace id", jSpanCtx.TraceID(), "span id", jSpanCtx.SpanID(),
```
Comment (alsamylkin) [inline modules/uhttp/filters.go]: Can we test it?

File: internal/fxcontext/context_test.go [hunk 5]
```diff
@@ -24,6 +24,8 @@ import (
 	gcontext "context"
 	"testing"
 
+	"golang.org/x/net/context"
```
Comment (anuptalwalkar) [inline internal/fxcontext/context_test.go]: use gcontext

File: modules/uhttp/filters.go (hunk 3 above)
Comment (madhuravi) [inline modules/uhttp/filters.go]: What if its not a jaeger tracer? uberfx can be used with other types of tracers.

File: modules/uhttp/filters.go (hunk 4 above)
Comment (madhuravi) [inline modules/uhttp/filters.go]: I mentioned to Anup offline. I am writing tests, expect an update soon.

File: modules/uhttp/filters.go (hunk 3 above)
Comment (alsamylkin) [inline modules/uhttp/filters.go]: I meant this:
```go
if jSpanCtx, ok := span.Context().(jaeger.SpanContext); ok {
    traceLogger := fxctx.Logger().With(
 				"trace id", jSpanCtx.TraceID(), "span id", jSpanCtx.SpanID(),
 				)
    fxctx = fxctx.WithLogger(traceLogger)
}
```
Should be ok?

File: internal/fxcontext/context.go (hunk 2 above)
Comment (madhuravi) [inline internal/fxcontext/context.go]: Store can be created without a logger. That doesn't make much sense right now but in future once we add more things to the store object for eg. auth etc. It is possible to have one set and not the other.
